src/components/createActivityForm/CreateActivityForm.js

```diff
@@ -0,0 +1,158 @@
+import { nanoid } from 'nanoid'
+import styled from 'styled-components'
+
+export default function CreateActivityForm({ activities, setActivities }) {
+  function createNewActivity({
+    name,
+    description,
+    street,
+    zipCode,
+    city,
+    country,
+    openingHours,
+    website,
+    isFreeOfCharge,
+  }) {
+    const newActivity = [
+      ...activities,
+      {
+        id: nanoid(),
+        name: name,
+        description: description,
+        street: street,
+        city: city,
+        zipCode: zipCode,
+        country: country,
+        openingHours: openingHours,
+        website: website,
+        isFreeOfCharge: isFreeOfCharge,
+      },
+    ]
+    setActivities(newActivity)
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    const form = event.target
+    const {
+      name,
+      description,
+      street,
+      zipCode,
+      city,
+      country,
+      openingHours,
+      website,
+      isFreeOfCharge,
+    } = form.elements
+
+    createNewActivity({
+      name: name.value,
+      description: description.value,
+      street: street.value,
+      city: city.value,
+      zipCode: zipCode.value,
+      country: country.value,
+      openingHours: openingHours.value,
+      website: website.value,
+      isFreeOfCharge: JSON.parse(isFreeOfCharge.value),
+    })
+    form.reset()
+  }
+
+  return (
+    <CreateForm
+      onSubmit={event => {
+        handleSubmit(event)
+      }}
```

Comment:
```suggestion
      onSubmit={handleSubmit}
```
No need to wrap it into another function call